app/controllers/application_controller.rb

```diff
@@ -5,8 +5,12 @@ class ApplicationController < ActionController::API
   private
 
   def not_found_response(exception)
-    render json: ErrorSerializer.new(ErrorMessage.new(exception.message, 404))
-    .serialize_json, status: :not_found
+    if exception.id.class == Integer
+      render json: ErrorSerializer.new(ErrorMessage.new(exception.message, 404))
```

Comment:
So to share another way that we could check for an Integer would be exception.id.is_a?(Integer)